posit-dev/positron
# Update tree-sitter-r.wasm and make queries work

Addresses #2632

This gets us to a new and happy place: using the same tree-sitter grammar for R in the front and back ends 🎉 

Related to https://github.com/posit-dev/amalthea/pull/290

I made the smallest changes possible to the existing queries to adapt to the new R grammar. Further changes are coming in a separate PR (or several) around these queries.

### Approach

@jennybc generated this `tree-sitter-r.wasm` with:

* tree-sitter cli v0.21.0. Although the latest tree-sitter is v0.22.2, @davisvaughan advises that we are effectively pinned at a lower version for the time being.
* `next` branch of https://github.com/r-lib/tree-sitter-r at https://github.com/r-lib/tree-sitter-r/commit/11ab554cafa88f3482a2dfd4fb6c93e638a22d0e
*  Literally, I used `tree-sitter generate && tree-sitter build-wasm` in a suitable checkout of `r-lib/tree-sitter-r`.

I think we probably want to develop a bit more ceremony, record-keeping, and (maybe) automation around placing `tree-sitter-r.wasm` in positron-r, but I'll leave that for the future.

### QA Notes

Open an R source package in positron and run its tests using the test explorer (the flask in the activity bar). That should work. Specifically, it should work as well (or as poorly) as before this PR. There are some known, open issues for the test explorer. I included the glue package (https://github.com/tidyverse/glue) in my manual tests because it uses both `test_that()` and `describe(it())` tests.
